src/lib.rs

```diff
@@ -152,6 +214,8 @@ impl<K: Hash + Eq, V: Weighable, S: BuildHasher> WeightCache<K, V, S> {
     pub fn put(&mut self, key: K, value: V) -> Result<(), ValueTooBigError> {
         let weight = V::measure(&value);
         if weight > self.max {
+            #[cfg(feature = "metrics")]
+            self.metrics.inserts_fail.inc();
```

Comment:
Not sure we need this. It will always be 0 for a normal program, and you do get an error.

Reply:
Well, it shows you that the cache is not correctly sized for your data, I think that's valuable.